doc/i2p.md

```diff
@@ -47,9 +53,34 @@ In a typical situation, this suffices:
 bitcoind -i2psam=127.0.0.1:7656
 ```
 
-The first time Bitcoin Core connects to the I2P router, its I2P address (and
-corresponding private key) will be automatically generated and saved in a file
-named `i2p_private_key` in the Bitcoin Core data directory.
+The first time Bitcoin Core connects to the I2P router, if a persistent I2P
+address is to be used, then it (and its corresponding private key) will be
+automatically generated and saved in a file named `i2p_private_key` in the
+Bitcoin Core data directory. A persistent I2P address is used for accepting
+incoming connections (`-i2pacceptincoming`) and for making outgoing connections
+if `-i2ptransientout` is not used.
+
+## Persistent vs transient I2P addresses
+
+In I2P connections, the host that receives the connection knows the I2P address
+of the connection initiator. This is unlike the Tor network where the recipient
+does not know who is connecting to them, not even the initiator's Tor address.
+
+So, even if an I2P node is not accepting incoming connections, they are known
+to other nodes by their outgoing I2P address. This creates an opportunity to
+white-list given nodes or treat them differently based on their I2P address.
+However, this also creates an opportunity to fingerprint or analyze a given node
+because it always uses the same I2P address when it connects to other nodes. If
+this is undesirable, then a node operator can use `-i2ptransientout` to generate
+a transient (disposable), one-time I2P address for each new outgoing connection.
+That address is never going to be reused again, not even if reconnecting to the
+same peer later.
+
+`-i2ptransientout` only affects addresses for outgoing connections and not the
+address for accepting incoming connections (if `-i2pacceptincoming` is given).
+This is because the address for accepting connections is advertised to the
+Bitcoin network and it takes time for it to propagate and be known by other
+nodes. So, it does not make sense to change it often - it must be persistent.
```

Comment:
How often can this be changed if a user wants to do it?

Reply:
It is hard to give a concrete answer but it could be _days_ before one's address is well propagated through the network.

You can measure for yourself in your environment: delete `~/.bitcoin/i2p_private_key` and restart `bitcoind` - it will generate a fresh I2P address for listening. Then monitor the incoming connections with something like `bitcoin-cli getpeerinfo |jq 'map(select(.network == "i2p" and .inbound)) |length'` or by enabling `-debug=net` and grepping `debug.log` for something like `connection from.*i2p accepted`.